bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/DataFlowDiagramConverter.java

```diff
@@ -0,0 +1,432 @@
+            for (Label label : labelType.getLabel()) {
+                values.add(new Value(label.getId(), label.getEntityName()));
+            }
+            labelTypes.add(new WebEditorLabelType(labelType.getId(), labelType.getEntityName(), values));
+        }
+    }
+
+    private WebEditorDfd processDfd(DataFlowDiagram dataFlowDiagram, DataDictionary dataDictionary) {
+        List<Child> children = new ArrayList<>();
+        List<WebEditorLabelType> labelTypes = new ArrayList<>();
+
+        createLabelTypesAndValues(labelTypes, dataDictionary);
+
+        createFlows(dataFlowDiagram, children);
+
+        createNodes(dataFlowDiagram, children);
+
+        return new WebEditorDfd(new Model("graph", "root", children), labelTypes);
+    }
+
+    private void createNodes(DataFlowDiagram dataFlowDiagram, List<Child> children) {
+        for (Node node : dataFlowDiagram.getNodes()) {
+            String text = node.getEntityName();
+            String id = node.getId();
+            String type;
+            if (node instanceof Process) {
+                type = "node:function";
+            } else if (node instanceof Store) {
+                type = "node:storage";
+            } else if (node instanceof External) {
+                type = "node:input-output";
+            } else {
+                type = "error";
+            }
+
+            List<WebEditorLabel> labels = new ArrayList<>();
+            for (Label label : node.getProperties()) {
+                String labelTypeId = ((LabelType) label.eContainer()).getId();
+                String labelId = label.getId();
+                labels.add(new WebEditorLabel(labelTypeId, labelId));
+            }
+
+            List<Port> ports = new ArrayList<>();
+
+            for (Pin pin : node.getBehaviour().getInPin()) {
+                ports.add(new Port(null, pin.getId(), "port:dfd-input", new ArrayList<>()));
+            }
+
+            Map<Pin, List<AbstractAssignment>> mapPinToAssignments = mapping(node);
+
+            for (Pin pin : node.getBehaviour().getOutPin()) {
+                String behaviour = createBehaviourString(mapPinToAssignments.get(pin));
+                ports.add(new Port(behaviour, pin.getId(), "port:dfd-output", new ArrayList<>()));
+            }
+
+            children.add(new Child(text, labels, ports, id, type, null, null, new ArrayList<>()));
+        }
+    }
+
+    private void createFlows(DataFlowDiagram dataFlowDiagram, List<Child> children) {
+        for (Flow flow : dataFlowDiagram.getFlows()) {
+            String id = flow.getId();
+            String type = "edge:arrow";
+            String sourceId = flow.getSourcePin().getId();
+            String targetId = flow.getDestinationPin().getId();
+            String text = flow.getEntityName();
+            mapInputPinToFlowName.put(flow.getDestinationPin(), text);
+            children.add(new Child(text, null, null, id, type, sourceId, targetId, new ArrayList<>()));
+        }
```

Comment:
Extract the iteration over one flow into a separate method (it would be helpful for debugging the creation of children here)